src/components/safe-messages/MsgModal/index.tsx

```diff
@@ -102,22 +105,45 @@ const MsgModal = ({
     setSubmitError(undefined)
 
     try {
-      if (requestId && !alreadyProposedMessage) {
+      // When collecting the first signature
+      if (requestId && !ongoingMessage) {
         await dispatchSafeMsgProposal({ onboard, safe, message: decodedMessage, requestId, safeAppId })
+        const message = await dispatchPreparedSignature(safe.chainId, safeMessageHash, onClose, requestId)
+        setOngoingMessage(message)
       } else {
         await dispatchSafeMsgConfirmation({ onboard, safe, message: decodedMessage, requestId })
+        dispatchPreparedSignature(safe.chainId, safeMessageHash, onClose, requestId)
```

Comment:
Only dispatching when we sign will keep users stuck if they already signed  the message.

For instance if I login, logout and login again fast on opensea I will get the same login message to sign. Our previous solution fixed that.

This will also not submit the signature to the dapp if the last signer is a different owner in a different browser.

Reply:
Actually the polling does not work at all for me when I sign in different browsers:
![Screenshot 2023-04-19 at 14 54 45](https://user-images.githubusercontent.com/2670790/233081324-b5bfa450-8021-40c6-9d0a-b56d74a9ff38.png)
This screenshot shows the same message in 2 browsers. On the left the initial signer (with WalletConnect app open).
After adding the second signature (see window on the right) the message left still only has 1/2 signatures.

Reply:
I suggest moving it into a useEffect like this for instance
```ts
  useEffect(() => {
    if (ongoingMessage?.preparedSignature) {
      dispatchPreparedSignature(safe.chainId, safeMessageHash, onClose, requestId)
    }
  }, [ongoingMessage, safe.chainId, safeMessageHash, onClose, requestId])
```